### PR TITLE
Add methods for `adb reverse`

### PR DESCRIPTION
**Pull request by NotWoods**
Added methods for `adb reverse`, a command similar to `adb forward` that lets you reverse a socket connection.

### CODE REVIEW

File: test/unit/adb-commands-specs.js [hunk 1]
```diff
@@ -541,7 +541,23 @@ describe('adb commands', withMocks({adb, logcat, teen_process, net}, function (m
...
+      });
+    });
+    describe('reversePort', function () {
+      const sysPort = 12345,
+            devicePort = 54321;
+      it('reversePort should call shell with correct args', async function () {
+        mocks.adb.expects('adbExec')
+          .once().withExactArgs(['reverse', `tcp:${devicePort}`, `tcp:${sysPort}`])
+          .returns('');
+        await adb.forwardPort(devicePort, sysPort);
```
Comment (imurchie) [inline test/unit/adb-commands-specs.js]: Should this be calling `adb.reversePort()`?

File: test/unit/adb-commands-specs.js [hunk 2]
```diff
@@ -541,7 +541,23 @@ describe('adb commands', withMocks({adb, logcat, teen_process, net}, function (m
...
+      });
+      it('removePortReverse should call shell with correct args', async function () {
+        mocks.adb.expects('adbExec')
+            .once().withExactArgs(['reverse', `--remove`, `tcp:${devicePort}`])
+            .returns('');
+        await adb.removePortForward(devicePort);
```
Comment (imurchie) [inline test/unit/adb-commands-specs.js]: Should this be calling `adb.removePortReverse()?`

File: lib/tools/adb-commands.js [hunk 3]
```diff
@@ -1015,6 +1015,39 @@ methods.removePortForward = async function (systemPort) {
...
 };
 
+/**
+ * Get TCP port forwarding with adb on the device under test.
+ * @return {Array.<String>} The output of the corresponding adb command. An array contains each forwarding line of output
+ */
+methods.getReverseList = async function () {
+  log.debug(`List reverse forwarding ports`);
+  let connections = await this.adbExec(['reverse', '--list']);
+  return connections.split('\n');
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: use os.EOL instead to be consistent on Windows

File: lib/tools/adb-commands.js [hunk 4]
```diff
@@ -1015,6 +1015,39 @@ methods.removePortForward = async function (systemPort) {
   await this.adbExec(['forward', `--remove`, `tcp:${systemPort}`]);
 };
 
+/**
+ * Get TCP port forwarding with adb on the device under test.
+ * @return {Array.<String>} The output of the corresponding adb command. An array contains each forwarding line of output
+ */
+methods.getReverseList = async function () {
+  log.debug(`List reverse forwarding ports`);
+  let connections = await this.adbExec(['reverse', '--list']);
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: const

File: lib/tools/adb-commands.js [hunk 5]
```diff
@@ -1015,6 +1015,39 @@ methods.removePortForward = async function (systemPort) {
...
+
+/**
+ * Setup TCP port forwarding with adb on the device under test.
+ *
+ * @param {string|number} devicePort - The number of the remote device port.
+ * @param {string|number} systemPort - The number of the local system port.
+ */
+methods.reversePort = async function (devicePort, systemPort) {
+  log.debug(`Forwarding device: ${devicePort} to system: ${systemPort}`);
+  await this.adbExec(['reverse', `tcp:${devicePort}`, `tcp:${systemPort}`]);
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: shall we hardcode `tcp`?

File: lib/tools/adb-commands.js [hunk 6]
```diff
@@ -1021,12 +1022,13 @@ methods.removePortForward = async function (systemPort) {
  */
 methods.getReverseList = async function () {
   log.debug(`List reverse forwarding ports`);
-  let connections = await this.adbExec(['reverse', '--list']);
-  return connections.split('\n');
+  const connections = await this.adbExec(['reverse', '--list']);
+  return connections.split(EOL);
```
Comment (mykola-mokhnach) [inline lib/tools/adb-commands.js]: shall we make sure there are no empty lines here if trimmed?

`return connections.split(EOL).filter((line) => !!line.trim())`

Comment (mykola-mokhnach) [reply lib/tools/adb-commands.js]: same comment to the other method

File: test/unit/adb-commands-specs.js [hunk 7]
```diff
@@ -541,7 +541,23 @@ describe('adb commands', withMocks({adb, logcat, teen_process, net}, function (m
...
+        await adb.removePortForward(sysPort);
+      });
+    });
+    describe('reversePort', function () {
+      const sysPort = 12345,
+            devicePort = 54321;
+      it('reversePort should call shell with correct args', async function () {
+        mocks.adb.expects('adbExec')
+          .once().withExactArgs(['reverse', `tcp:${devicePort}`, `tcp:${sysPort}`])
+          .returns('');
```
Comment (mykola-mokhnach) [inline test/unit/adb-commands-specs.js]: should we also check here, that the output is properly parsed to a list of strings?

**Comment by imurchie** [comment]
Thanks for this work!

**Pull request merged**